### PR TITLE
[Merged by Bors] - chore: Split `data.set.pairwise`

### PR DESCRIPTION
**Pull request by YaelDillies**
Match https://github.com/leanprover-community/mathlib/pull/17880

The new import of `Mathlib.Data.Set.Lattice` in `Mathlib.Data.Finset.Basic` was implied transitively from tactic imports present in Lean 3.

---

- [x] depends on: #1619

There's no link to the port dashboard because the file was renamed.

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

File: Mathlib/Data/Finset/Basic.lean [hunk 1]
```diff
@@ -9,6 +9,7 @@ Authors: Leonardo de Moura, Jeremy Avigad, Minchao Wu, Mario Carneiro
 ! if you have ported upstream changes.
 -/
 import Mathlib.Data.Multiset.FinsetOps
+import Mathlib.Data.Set.Lattice
```
Comment (eric-wieser) [inline Mathlib/Data/Finset/Basic.lean:12]: Why didn't this change in mathlib3?

File: Mathlib/Order/Antichain.lean [hunk 2]
```diff
@@ -4,11 +4,14 @@ Released under Apache 2.0 license as described in the file LICENSE.
 Authors: Yaël Dillies
 
 ! This file was ported from Lean 3 source module order.antichain
-! leanprover-community/mathlib commit 207cfac9fcd06138865b5d04f7091e46d9320432
+! leanprover-community/mathlib commit c227d107bbada5d0d9d20287e3282c0a7f1651a0
```
Comment (eric-wieser) [inline Mathlib/Order/Antichain.lean:7]: You need to port https://github.com/leanprover-community/mathlib/pull/18636 first

File: Mathlib/Data/Set/Pairwise/Basic.lean [hunk 3]
```diff
@@ -3,22 +3,22 @@ Copyright (c) 2017 Johannes Hölzl. All rights reserved.
 Released under Apache 2.0 license as described in the file LICENSE.
 Authors: Johannes Hölzl
 
-! This file was ported from Lean 3 source module data.set.pairwise
-! leanprover-community/mathlib commit 207cfac9fcd06138865b5d04f7091e46d9320432
+! This file was ported from Lean 3 source module data.set.pairwise.basic
```
Comment (eric-wieser) [inline Mathlib/Data/Set/Pairwise/Basic.lean:6]: Ugh, renaming this file was a mistake; the port dashboard doesn't track renames well.

File: Mathlib/Order/Antichain.lean (hunk 2 above)
Comment (Parcly-Taxel) [reply Mathlib/Order/Antichain.lean:7]: #3125 (nice perfect power)

File: Mathlib/Data/Finset/Basic.lean (hunk 1 above)
[the remark this answers is not in the document]
Comment (eric-wieser) [reply Mathlib/Data/Finset/Basic.lean:12]: We should try removing it again with the intermediate backport now in place

File: Mathlib/Data/Set/Pointwise/SMul.lean [hunk 4]
```diff
@@ -4,12 +4,12 @@ Released under Apache 2.0 license as described in the file LICENSE.
 Authors: Johan Commelin, Floris van Doorn
 
 ! This file was ported from Lean 3 source module data.set.pointwise.smul
-! leanprover-community/mathlib commit 7b78d1776212a91ecc94cf601f83bdcc46b04213
+! leanprover-community/mathlib commit c227d107bbada5d0d9d20287e3282c0a7f1651a0
```
Comment (eric-wieser) [inline Mathlib/Data/Set/Pointwise/SMul.lean:7]: You need to port #17961 first

**Comment by semorrison** [comment]
This PR/issue depends on:
* leanprover-community/mathlib4#1619
By **[Dependent Issues](https://github.com/z0al/dependent-issues)** (🤖). Happy coding!

**Comment by semorrison** [comment]
This PR/issue depends on:
* ~~leanprover-community/mathlib4#1619~~
By **[Dependent Issues](https://github.com/z0al/dependent-issues)** (🤖). Happy coding!

**Review by eric-wieser — APPROVED** [review]
bors merge

**Pull request closed without merging**